### DISCUSSION TITLE
How to close done？

### DISCUSSION THREAD

**Discussion by askfiy** [hrsh7th/nvim-cmp#775] [Q&A]
I want the completion to be written to my buffer only after I press <cr> instead of every time I select, can this be done? what should I do?

**Comment by askfiy** [comment]
Select, previous or next. The text content also changes, I want it to change after pressing <cr>.

**Comment by askfiy** [comment]
Or when I select, the content doesn't autofill after I press to close the popup

**Comment by gegoune** [comment]
That's not an issue, you should ask in Discussions. Sounds like you don't want `select` behaviour. It's described in documentation.

**Comment by askfiy** [comment]
> That's not an issue, you should ask in Discussions. Sounds like you don't want `select` behaviour. It's described in documentation.

Can you give me more detailed help? Thank you very much

**Comment by gegoune** [comment]
I am not sure exactly what you are trying to achieve. Please try to describe your current and desired state of affairs in more detail.


@hrsh7th (please move it to discussions)

**Reply by askfiy** [reply]
In vscode, up and down selection suggestions are not written to the buffer:

![vscode](https://user-images.githubusercontent.com/81478335/152463279-2d16b83d-7aca-4b04-a0ba-7a45bec22ce0.gif)

But in vim, the suggestion is written when selecting a suggestion:

![vim](https://user-images.githubusercontent.com/81478335/152463429-347ce8cf-d515-40f3-bcb8-31ae714ef52e.gif)

I need the functionality in vscode

**Reply by Shougo** [reply]
You can change the behavior by `cmp.SelectBehavior`.
`cmp.SelectBehavior.Insert` or `cmp.SelectBehavior.Select`.

```
*cmp.select_next_item* (option: { behavior = cmp.SelectBehavior })
  Select next item.

*cmp.select_prev_item* (option: { behavior = cmp.SelectBehavior })*
  Select prev item.
```
